apps/pace/app.ts

```diff
@@ -0,0 +1,225 @@
+const exs = require("exstats").getStats(
+  ["dist", "pacec"],
+  {
+   notify: {
+      dist: {
+        increment: 1000,
+      },
+    },
+  },
+);
+const S = require("Storage");
+
+let drawTimeout: TimeoutId | undefined;
+
+type Dist = number & { brand: 'dist' };
+type Time = number & { brand: 'time' };
+
+type Split = {
+  dist: Dist,
+  time: Time,
+};
+
+const splits: Split[] = []; // times
+let splitOffset = 0, splitOffsetPx = 0;
+
+const GPS_TIMEOUT_MS = 30000;
+
+const layout = new Layout({
+  type: "v",
+  c: [
+    {
+      type: "txt",
+      font: "6x8:2",
+      label: "Pace",
+      id: "paceLabel",
+      pad: 4
+    },
+    {
+      type: "txt",
+      font: "Vector:40",
+      label: "",
+      id: "pace",
+      halign: 0
+    },
+    {
+      type: "txt",
+      font: "6x8:2",
+      label: "Time",
+      id: "timeLabel",
+      pad: 4
+    },
+    {
+      type: "txt",
+      font: "Vector:40",
+      label: "",
+      id: "time",
+      halign: 0
+    },
+  ]
+}, {
+  lazy: true
+});
+
+const draw = () => {
+  if (!exs.state.active) {
+    // no draw-timeout here, only on user interaction
+    drawSplits();
+    return;
+  }
+
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(draw, 1000);
+
+  const now = Date.now();
+
+  let pace: string;
+  if ("time" in exs.state.thisGPS
+  && now - (exs.state.thisGPS.time as unknown as number) < GPS_TIMEOUT_MS)
+  {
+    pace = exs.stats.pacec.getString()
+  }else{
+    pace = "No GPS";
+  }
+
+  layout["time"]!.label = formatDuration(exs.state.duration);
+  layout["pace"]!.label = pace;
+  layout.render();
+};
+
+const pad2 = (n: number) => `0${n}`.substr(-2);
+
+const formatDuration = (ms: number) => {
+  const tm = time_utils.decodeTime(ms);
+  if(tm.h)
+    return `${tm.h}:${pad2(tm.m)}:${pad2(tm.s)}`;
+  return `${pad2(tm.m)}:${pad2(tm.s)}`;
+};
+
+// divide by actual distance, scale to milliseconds
+const calculatePace = (split: Split) => formatDuration(split.time / split.dist * 1000);
+
+const drawSplits = () => {
+  g.clearRect(Bangle.appRect);
+
+  const barSize = 20;
+  const barSpacing = 10;
+  const w = g.getWidth();
+  const h = g.getHeight();
+
+  const max = splits.reduce((a, s) => Math.max(a, s.time), 0);
+
+  g.setFont("6x8", 2).setFontAlign(-1, -1);
+
+  let i = 0;
+  for(; ; i++) {
+    const split = splits[i + splitOffset];
+    if (split == null) break;
+
+    const y = Bangle.appRect.y + i * (barSize + barSpacing) + barSpacing / 2;
+    if (y > h) break;
+
+    const size = w * split.time / max; // Scale bar height based on pace
+    g.setColor("#00f").fillRect(0, y, size, y + barSize);
+
+    const splitPace = calculatePace(split); // Pace per km
+    drawSplit(i, y, splitPace);
+  }
+
+  const pace = exs.stats.pacec.getString();
+
+  const y = Bangle.appRect.y + i * (barSize + barSpacing) + barSpacing / 2;
+  drawSplit(i, y, pace);
+};
+
+const drawSplit = (i: number, y: number, pace: number | string) => {
+  g
+    .setColor("#fff")
```

Comment:
Should use `g.theme.fg`, right (or similar)? On Light BW theme it's white on white for me.

Reply:
Yes, good spot!